lib/Emulation.cpp

```diff
@@ -224,9 +224,9 @@ void Emulation::receiveData(const char* text, int length)
      * U+10FFFF
      * https://unicodebook.readthedocs.io/unicode_encodings.html#surrogates
      */
-    QString str = QString::fromUtf8(text, length);
-    auto encoded = _fromUtf8(str);
-    std::wstring unicodeText = encoded.data.toStdWString(); 
+    QByteArray ba(text, length);
+    QString str = _toUtf16(ba);
```

Comment:
nit: if you wanted to be a lil more specific here[^1], even though the [examples in the doc](https://doc.qt.io/qt-6/qstringdecoder.html#details) support your current version of decoding, you could use:
```diff
-    QString str = _toUtf16(ba);
+    QString str = _toUtf16.decode(ba);
```
doc: [here](https://doc.qt.io/qt-6/qstringdecoder.html#decode-1)

and if you wanted to be even more explicit, can also do:
```diff
-  _toUtf16(QStringConverter::Utf8)
+  _toUtf16(QStringConverter::Utf8, QStringConverter::Flag::Default)
```
docs: [constructor](https://doc.qt.io/qt-6/qstringdecoder.html#QStringDecoder-1) and [flags](https://doc.qt.io/qt-6/qstringconverter.html#Flag-enum)

I've tested both methods(above) to compile and work correctly. Locally, I'll be using this variant of the patch.

imho, the patch as it is, seems to me, to work the exact same way as 1.4.0 [worked](https://github.com/lxqt/qtermwidget/pull/532/files#diff-0ce76997398f4a2c9df299136c2a5e7d44cfc4a410841b18ff2c5919b4032b84L250-L251). So I don't see why/when it wouldn't work, but then again, I'm noob at this, so who knows.

Meanwhile it passed "testing" in `btop`, `nload` and `top` which are the only things that I use `qterminal` for. 

[^1]: (I personally have a dislike for the implicit things(musta gotten bit too many times, + hurt readability and ended up wasting time to offset it) but that's just me)

Reply:
> if you wanted to be a lil more specific here...

They're exactly the same thing.

> and if you wanted to be even more explicit,...

Not to this extent ;)

Reply:
oh, alright, I only suggested it because I saw you do this:
```diff
-  _toUtf16(QStringEncoder::Utf8)
+  _toUtf16(QStringConverter::Utf8)
```
and I figured well why did you stop there?
but actually I thought it was _toUtf16(QString**Decoder**::Utf8) before, so the change makes more sense. I thought you were just being more explicit since they all the same thing anyway.

Nevermind, forget I mentioned anything :)

Reply:
> I only suggested it because I saw you do this:

Yes, I guessed that :) I did so only to prevent confusion for code readers (we have a decoder now, not an encoder).